src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int BORDER = 30;
+
+    public Figures getRandomFigure() {
+        Random random = new Random();
+        ColorSupplier supplier = new ColorSupplier();
+        RightTriangle triangle = new RightTriangle("right triangle",
+                supplier.getRandomColor(),
+                random.nextInt(BORDER), random.nextInt(BORDER));
+        Rectangle rectangle = new Rectangle("rectangle",
+                supplier.getRandomColor(),
+                 random.nextInt(BORDER), random.nextInt(BORDER));
+        Square square = new Square("square",
+                supplier.getRandomColor(), random.nextInt(BORDER));
+        Circle circle = new Circle("circle",
+                supplier.getRandomColor(), random.nextInt(BORDER));
+        Trapezoid trapezoid = new Trapezoid("isosceles trapezoid",
+                supplier.getRandomColor(), random.nextInt(BORDER),
+                random.nextInt(BORDER), random.nextInt(BORDER));
+        Figures[] array = new Figures[]{triangle, rectangle, square, circle, trapezoid};
+        int index = random.nextInt(array.length);
+        return array[index];

```

Comment:
can we refactor it and use switch/case here?

Reply:
yes, we can use switch with random number
I can leave this variant or I have to corect its ??
this variant quite useful with using arrays random or not ??

Reply:
+ in my variant i make less new objects

Reply:
you have to change it to witch/case because at the moment you're creating too many objects, all your random figures and an array will be created when you only need one random figure.
But if you use switch case you will create only one object of a random figure, so there will be less memory used

Reply:
thanks for your explaining